java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -315,12 +306,13 @@ public void visitTypeInsn(int opcode, String type) {
     public void visitJumpInsn(int opcode, Label label) {
       if(opcode == GOTO || opcode == JSR) {
         currentBlock.terminator = new Instruction(opcode);
-        blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+        currentBlock.successors = Collections.singletonList(blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor()));
         return;
       }
-      blockByLabel.computeIfAbsent(label, l -> currentBlock.createTrueSuccessor());
+      currentBlock.trueBlock = blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
```

Comment:
If the label is already associated with the block, returned block will be set as a `trueBlock` but it will be not part of `successors` of `currentBlock`